framework/src/transfers/MultiAppNearestNodeTransfer.C

```diff
@@ -159,23 +164,43 @@ MultiAppNearestNodeTransfer::execute()
                from0 += froms_per_proc[i_proc], i_proc++)
           {
             bool qp_found = false;
+
             for (unsigned int i_from = from0; i_from < from0 + froms_per_proc[i_proc] && !qp_found;
                  i_from++)
             {
+
               Real distance = bboxMinDistance(*node, bboxes[i_from]);
-              if (distance < nearest_max_distance || bboxes[i_from].contains_point(*node))
+
+              if (distance <= nearest_max_distance || bboxes[i_from].contains_point(*node))
               {
                 std::pair<unsigned int, unsigned int> key(i_to, node->id());
                 node_index_map[i_proc][key] = outgoing_qps[i_proc].size();
                 outgoing_qps[i_proc].push_back(*node + _to_positions[i_to]);
                 qp_found = true;
+                local_nodes_found.insert(node);
               }
             }
           }
         }
+
+        // By the time we get to here, we should have found at least
+        // one candidate BoundingBox for every node in the
+        // target_local_nodes array that has dofs for the current
+        // variable in the current System.
+        for (const auto & node : target_local_nodes)
+          if (node->n_dofs(sys_num, var_num) && !local_nodes_found.count(node))
+            mooseError("No candidate BoundingBoxes found for node ",
```

Comment:
Needs `name()` here. When you have multiple transfers, you want to know which one failed.

Reply:
Fixed